src/eradiate/scenes/phase/_blend.py

```diff
@@ -163,33 +181,27 @@ def _gridvolume_transform(self) -> "mitsuba.core.ScalarTransform4f":
 
     def kernel_dict(self, ctx: KernelDictContext) -> KernelDict:
         # Build kernel dict recursively
-        weights = self.weights[0, ...]
+
+        # Weight of all nested phase function but the first one
+        # (in the case where there are two nested phase functions,
+        # this is the weight of the second nested phase function):
+        weight = self.weights[1:, ...].sum(axis=0)
         phase_dict_1 = onedict_value(self.components[0].kernel_dict(ctx))
 
         if len(self.components) == 2:
             phase_dict_2 = onedict_value(self.components[1].kernel_dict(ctx))
 
         else:
-            sub_weights = self.weights[1:, ...]
-            sum_sub_weights = np.sum(self.weights[1:, ...], axis=0)
-
-            # Zero divisions might occur when the last component has zero
-            # scattering coefficient. In order to prevent those to happen, we
-            # detect them and assign 1 to the last probability.
-            marginal_weights = np.divide(
-                sub_weights,
-                sum_sub_weights,
-                where=sum_sub_weights != 0.0,
-                out=np.ones_like(sub_weights),
-            )
-
+            # All nested phase function but the first one are aggreated
+            # in a blend phase function, that is nested into the current
+            # blend phase function.
```

Comment:
Be more concise, the code explains this already:
```suggestion
            # The phase function corresponding to remaining components
            # is built recursively
```